### ISSUE TITLE
Hammerhead is changing the behavior of the page

### ISSUE THREAD

**Issue by kolman** [DevExpress/testcafe#1973]
### Are you requesting a feature or reporting a bug?
Bug

### What is the current behavior?
When server returns HTTP status 500, hammerhead throws it's own TypeError, messing with the application's own error handling. If the application handles the HTTP 500 error, it changes it's behavior.

### What is the expected behavior?
I am not sure what exactly hammerhead does (monitoring the network requests?) but it should not replace the original exception with type error. Most likely the problem is hammerhead's src/client/sandbox/fetch.js, line 87:

```js
            args[0] = function (response) {
                if (response.status === 500)
                    throw new TypeError();
```

### How would you reproduce the current behavior (if this is a bug)?
We are using "whatwg-fetch": "2.0.3" to do XHR, then check the response and throw an error:

```js
function checkStatus(response) {
  if (response.status >= 200 && response.status < 300) {
    return response;
  }
  const error = new Error(response.statusText);
  error.response = response;
  throw error;
}

const promise = fetch(...).then(checkStatus);

promise.catch(e => console.log(e.message)); // e is TypeError thrown by hammerhead, no message
```

### Specify your

* testcafe version: 0.18.3

**Comment by miherlosev** [comment]
Hi, @kolman 
Thanks for your report.
I've reproduced the problem and created a separate issue in the web proxy repository - https://github.com/DevExpress/testcafe-hammerhead/issues/1397

**Comment by LavrovArtem** [comment]
Hi @kolman,
The fix is published in `testcafe@0.18.6-dev20171222`.
